log-analyzer/features/org.wso2.carbon.la.apim.feature/src/main/resources/gadgets/LogAPIMArtifactDeleted/js/VizGrammar.js

```diff
@@ -0,0 +1,2300 @@
+            "range": config.colorScale
+        };
+
+        scales.push(colorScale);
+
+        var legends = [
+            {
+                "fill": "color",
+                "title": "Legend",
+                "offset": 10,
+                "properties": {
+                    "symbols": {
+                        "fillOpacity": {"value": 0.5},
+                        "stroke": {"value": "transparent"}
+                    }
+                }
+            }
+        ];
+
+
+        if (config.mode == "stack") {
+            var aggregateData = {
+                "name": "stack",
+                "source": config.title,
+                "transform": [
+                    {
+                        "type": "aggregate",
+                        "groupby": [this.metadata.names[config.x]],
+                        "summarize": [{"field": this.metadata.names[config.y], "ops": ["sum"]}]
+                    }
+                ]
+            };
+
+            dataTable.push(aggregateData);
+            yColumn = "sum_"+ this.metadata.names[config.y];
+            yDomain = "stack";
+
+        } else {
+            yColumn = this.metadata.names[config.y];
+            yDomain = config.title;
+        }
+
+        this.spec.legends = legends;
+    } else {
+        yColumn = this.metadata.names[config.y];
+        yDomain = config.title;
+    }
+
+    var xScale = {
+        "name": "x",
+        "type": "ordinal",
+        "range": xRange,
+        "domain": {"data":  config.title, "field": this.metadata.names[config.x]}
+    };
+
+    if (config.mode == "group") {
+        xScale.padding = 0.2;
+    }
+
+    var yScale = {
+        "name": "y",
+        "type": this.metadata.types[config.y],
+        "range": yRange,
+        "domain": {"data": yDomain, "field": yColumn}
+    };
+
+
+
+    scales.push(xScale);
+    scales.push(yScale);
+
+    var axes =  getXYAxes(config, xAxesType, "x", yAxesType, "y");
+
+    if (config.color != -1 && config.mode == "stack") {
+        marks.push(getStackBarMark(config, this.metadata));
+    } else if (config.color != -1 && config.mode == "group") {
+        marks.push(getGroupBarMark(config, this.metadata));
+    } else {
+        marks.push(getBarMark(config, this.metadata));
+    }
+
+    if (config.range) {
+        signals = getRangeSignals(config, signals);
+        marks = getRangeMark(config, marks);
+    }
+
+    this.spec.width = config.width;
+    this.spec.height = config.height;
+    this.spec.axes = axes;
+    this.spec.data = dataTable;
+    this.spec.scales = scales;
+    this.spec.padding = config.padding;
+    this.spec.marks = marks;
+    this.spec.signals = signals;
+
+    var specc = JSON.stringify(this.spec);
+};
+
+bar.prototype.draw = function(div, callbacks) {
+    if(this.config.maxLength != -1){
+        var dataset = this.spec.data[0].values;
+        var maxValue = this.config.maxLength;
+        if(dataset.length >= this.config.maxLength){
```

Comment:
Hi Chandana,
We have replace the Vizgrammer.js with Vizgrammer.min.js and relevant commits are in

https://github.com/wso2/shared-analytics/pull/126/files#diff-6ac780d61b2b6d470c8ff04bdeed3bcdR52

Thanks,
SajithD